All: Add Multi-factor authentication guide (#10661)
laurent22/joplin: readme/apps/mfa.md

# Multi-factor authentication

> You can now enable Multi-Factor Authentication (MFA) to your Joplin Account. MFA enhances your account security by requiring more than just a password to log in. 

Reviewer: ```suggestion
Multi-Factor Authentication (MFA), also known as Two Factor Authentication (2FA),  enhances your account security by requiring more than just a password to log in. 
```

> If you can't generate your authentication code because you have lost access to your smartphone, for example, you will need to use one of the recovery codes to log in.

Reviewer: Move under "Recovery codes" section

> ## Recovery Codes

Reviewer: ```suggestion
## Recovery codes
```

> When you enable MFA, Joplin Cloud will show you a list of codes, that you should save in somewhere safe. Each code can only be used once, but you can generate new codes, if necessary, from your profile page.

Reviewer: Move the "Recovery codes" section after "How to disable MFA"

## How to enable MFA

- [Access your profile](https://joplincloud.com/users/me)
- Find the *Multi-factor authentication* section
- Click the link to *Enable Multi-Factor Authentication*
- Follow the instructions shown in the page

> If successful you should see a notification and a new page with your recovery codes, **remember to save them**.

Reviewer: That line can be removed as we already say that in the Recovery codes section

## How to disable MFA

- [Access your profile](https://joplincloud.com/users/me)
- Find the *Multi-factor authentication* section
- Click the link to *I want to disable it*

> - Finish by entering your password

Reviewer: ```suggestion
- Click the link to *I want to disable it* and follow the instructions
```

## I forgot to copy my recovery codes, how do I find them?

- [Access your profile](https://joplincloud.com/users/me)
- Find the *Multi-factor authentication* section

> - Click the link to *Recovery Codes*

Reviewer: How is the link called? We should put the name as the grammar is not right here

Author: I'm not sure if I understand your suggestion.

![image](https://github.com/laurent22/joplin/assets/5051088/ad9f3e80-834e-451b-b1fc-cc824c6da7cb)

Author: @laurent22 I'm not sure what change is needed here, can you help me?

- Follow the instructions shown in the page

> You will only be able to see your recovery codes if you have MFA enabled

## How do I generate new recovery codes?

- [Access your profile](https://joplincloud.com/users/me)
- Find the *Multi-factor authentication* section

> - Click the link to *Recovery Codes*

Reviewer: Fix grammar

# Multi-factor authentication

> You can now enable Multi-Factor Authentication (MFA) to your Joplin Account.

Reviewer: For the whole document, please don't add line breaks in paragraphs. We generally don't rely on soft breaks and I don't know how scripts along the pipeline to build the website are going to handle them. So essentially each paragraph should be on a single line

MFA enhances your account security by requiring more than just a password to log in.
Joplin Cloud uses a method that generates a new code every 30s, this code needs 
to be entered on each new log in after MFA is enabled.

If you can't generate your time-based code because you have lost access to your 
smartphone, for example, you will need to use one of the recovery codes to log in.

## Recovery Codes

When you enable MFA, Joplin Cloud will show you a list of codes,that you should save in 
somewhere safe. Each code can only be used once, but you can generate new codes,
if necessary, from your profile page.

> ### How to enable MFA

Reviewer: All the `###` should be `##` (since they are not all a sub-section of `Recovery codes`)

codes, **remember to save them**.

### How to disable MFA

Access your profile by clicking your full name in the top bar. Find the 
*Multi-factor authentication* section, and there should be link to *I want to disable it*.

### I forgot to copy my recovery codes, how do I find them?

If you have enabled MFA, but forgot to save your recovery codes, you can retrieve 
them again by going to your profile and clicking the *Recovery Codes* link in the
*Multi-factor authentication* section.

Before viewing your recovery codes, you will be prompted to enter an authentication code
(generated by your app) or your password. Every time you access your recovery code, 
Joplin Cloud will send you an alert email.

In this page, you will find a button at the bottom that will generate new codes,
don't forget to copy the new codes if you opt to generate new ones.

### How to log in with my recovery code

If you need to make login with your recovery code, when you are prompt to enter
your authentication code in the login screen you will see a link called
*Use a recovery code*. In this screen you have a new input field for the recovery

> code.

Reviewer: This doc is well written but I wonder if we should aim to make it shorter. Partly because, well, users don't like to read the manual, and partly because it's more to maintain and translate. Have a look at at how it's done for example on Dropbox or OneDrive: https://help.dropbox.com/account-access/enable-two-step-verification https://support.microsoft.com/en-us/account-billing/how-to-use-two-step-verification-with-your-microsoft-account-c7910146-672f-01e9-50a0-93b4585e7eb4

It's essentially click here, then follow the instructions. That means our instructions during the process should be clear enough, so that we can keep the doc here to a minimum.

Also I think any process we describe (click here, click there), should be in bullet points.

I think the section titles you have are fine, just the content should be more concise, or bullet points
